### PR TITLE
BaseTiffReader - Index out of bounds exception 

### PR DESCRIPTION
**Pull request by dgault**
Issue was discovered while testing QA-17365 as part of https://www.openmicroscopy.org/community/viewtopic.php?f=13&t=8124&p=17519#p17519

An object of type String [] is read for the creation date of the file, but the array itself is empty leading to an index out of bounds exception.

To test:
- All unit tests and data repo tests must remain green
- Read image from QA-17365 using Bio-Formats and verify that an array out of bounds exception is thrown
- Apply the PR
- Reopen the file and verify that no exception is thrown

### CODE REVIEW

**Comment by sbesson** [comment]
Looks good and works as expected with the uploaded QA file. Is there some room for a unit test or would you rather open a configuration PR for the QA file?

**Comment by dgault** [comment]
We are actually missing the testing of invalid values from the unit tests so I will get those written for the base tiff tests as part of this PR.

**Comment by dgault** [comment]
The BaseTiffReader is not specifically covered by existing Unit Tests. I was originally going to add a full unit test for this abstract reader but it was becoming much larger than the scope of this PR. As such I have broken out only the necessary tests specific for this PR change (anything related to the creation date).

File: components/formats-bsd/test/loci/formats/utests/tiff/BaseTiffReaderMock.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package loci.formats.utests.tiff;
```
Comment (sbesson) [inline components/formats-bsd/test/loci/formats/utests/tiff/BaseTiffReaderMock.java]: Add standard BSD header ?

File: components/formats-bsd/test/loci/formats/utests/tiff/BaseTiffReaderTest.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
+/*
+ * #%L
+ * BSD implementations of Bio-Formats readers and writers
+ * %%
+ * Copyright (C) 2005 - 2016 Open Microscopy Environment:
```
Comment (sbesson) [inline components/formats-bsd/test/loci/formats/utests/tiff/BaseTiffReaderTest.java]: `s/2005 - 2016/2016` ?

**Pull request merged**